addons/account/models/account_move.py

```diff
@@ -3940,8 +3943,21 @@ def button_draft(self):
         self.write({'state': 'draft', 'is_move_sent': False})
 
     def button_cancel(self):
-        self.write({'auto_post': 'no', 'state': 'cancel'})
+        """Shortcut to move from posted to cancelled directly. This is useful for E-invoices that must not be changed
+        when sent to the government"""
+        moves_to_reset_draft = self.filtered(lambda x: x.state == 'posted')
+        if moves_to_reset_draft:
+            try:
+                moves_to_reset_draft.button_draft()
+            except UserError as e:
+                raise UserError(_("Some journal entries are posted. They need to be reset to draft first to cancel them. "
```

Comment:
I don't think this try except is necessary...
In 18.0 we simply try to reset to draft, and it may raise a user error.